### ISSUE TITLE
Bug when 2nd holding record is added to bib already processed

### ISSUE THREAD

**Issue by patrickzurek** [eXtensibleCatalog/xcmetadataservicestoolkit#446]
JIRA issue created by: rcook
Originally opened: 2012-02-28 03:42 PM

Issue body:
 (nt)

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-02-28 03:46 PM

Comment body:

Chris,

We believe there is a bug with held holdings.  The problem seems to be when a record is initially processed with one holdings records, and then a change subsequently occurs such that another holding record is associated with a bib already processed.  This may or may not have something to do with having the same location.

Let us know if it would be easier for you to mock this up or if you would like Ralph to help out.


From: Strong, Marcy 
Sent: Tuesday, February 28, 2012 8:54 AM
To: Bowen, Jennifer
Subject: RE: Multiple Sibley holdings

The XC issue is that they are in the Held file, which means the holdings record isnâ€™t attached to a bib record. It appears that the second MFHD with the same location doesnâ€™t attach to the bib like the first MFHD does.

From: Bowen, Jennifer 
Sent: Monday, February 27, 2012 12:45 PM
To: Strong, Marcy
Subject: RE: Multiple Sibley holdings

Oh, I see whatâ€™s going on here â€“ these are in the same location, but each copy of the LP has a difference accession number, so they need to have separate MFHDs.  So I donâ€™t think thereâ€™s a problem with the way they are cataloged.  Can you recall why XC was having a problem with them, or do we even know?

From: Strong, Marcy 
Sent: Tuesday, February 14, 2012 4:47 PM
To: Bowen, Jennifer
Subject: Multiple Sibley holdings

Hi Jennifer,

Here are some examples of the multiple holdings in Sibley (usually the SibRecrd location) found in the Held file:

Bib: 4779360 (held MFHD: 5382053)
Bib: 4774668 (held MFHD: 5382055)
Bib: 4774667 (held MFHD: 5382903)
Bib: 4772609 (held MFHD: 5383093) 

There are plenty more examples and they tend to be at the end of the held file. It does seem to be current practice at Sibley.

**Comment by patrickzurek** [comment]
JIRA Comment by user: Chris Delis (cedelis)
JIRA Timestamp: 2012-03-08 09:44 PM

Comment body:

Was able to reproduce this error using a very small subset of two records: one bib with two holdings record (each having 004 linking to the same bib).  attached.

Also, for good measure, it would be good to make sure MST can handle another test with two bibs and two holdings recors (each having 004s linking to the same two bibs).  attached.

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-03-08 10:36 PM

Comment body:

Also note, this was reported with an uplink from an 004 and the 2nd from the addition of an 014.

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-03-08 10:39 PM

Comment body:

Jennifer/Chris,
Is this use case ("Also, for good measure, it would be good to make sure MST can handle another test with two bibs and two holdings recors (each having 004s linking to the same two bibs)" a valid one?

**Comment by patrickzurek** [comment]
JIRA Comment by user: Chris Delis (cedelis)
JIRA Timestamp: 2012-03-08 10:42 PM

Comment body:

I don't know about the second use-case being valid in a real situation, but it should be valid I think from a purely objective view of parent-child relationships.  I'm not a library/marc expert by no means.  As for the distinction you made re: 004/014s.. it shouldn't matter; the solution should solve both; the use case I have triggers the bug; when fixing the bug it should also work with 014s (maybe we could add that as another test use-case).

**Comment by patrickzurek** [comment]
JIRA Coment by user: jbowen
JIRA Timestamp: 2012-03-09 01:58 PM

Comment body:

Yes, it is a valid use case, although very uncommon - don't know if we could find real records that would demonstrate this.  An example might be an old LP vinyl disc - old cataloging practice was to create a separate bib record for Side 1 and Side 2, so 2 bibs.  If you have 2 copies of that LP held in two different locations (2 MFHDS) your use case would occur.

**Comment by patrickzurek** [comment]
JIRA Comment by user: Chris Delis (cedelis)
JIRA Timestamp: 2012-03-13 06:23 PM

Comment body:

Boo...  I realized today that the two example sets included in this case are bogus and do not actually reproduce the issue.  I thought I was testing only the 004s, but there were 014s in the mfhds, too, which was the reason for them being held -- and not because of the issue presented in this fogbugz case.

Back to the drawing board... (i.e., need to reproduce this issue...)

--Chris

**Comment by patrickzurek** [comment]
JIRA Comment by user: Chris Delis (cedelis)
JIRA Timestamp: 2012-03-13 06:55 PM

Comment body:

Randy said:

"Chris,

We believe there is a bug with held holdings.  The problem seems to be when a record is initially processed with one holdings records, and then a change subsequently occurs such that another holding record is associated with a bib already processed.  This may or may not have something to do with having the same location.

Let us know if it would be easier for you to mock this up or if you would like Ralph to help out."

Randy,

I was unable to reproduce the issue given this scenario.  Perhaps I'm not interpreting you correctly.  So far, here are the two scenarios I created (which do NOT reproduce this issue, i.e., the MST behaves correctly):

Scenario 1: I harvested 4 total records: 2 bib, 2 mfhd.  Each mfhd had a different parent bib (mfhd1's parent was bib1; mfhd2's parent was bib2).  I modified mfhd2's parent so that it was now bib1.  The MST handled this relationship change correctly.  (No held records)

Scenario 2: Follow initial harvest as Scenario 1.  Instead of modifying mfhd2, instead I added a new mfhd3 with a parent of bib1.  The MST handled this new record (and its relationship) correctly.  (No held records) 

Assuming I didn't understand your hypothesis above, Randy, could you re-state it in a way that is similar to my above scenarios (i.e., much more explicitly)?

Thanks,
Chris

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-03-13 07:28 PM

Comment body:

I believe the following is the scenario:

bib1 processed, along with mfhd2.  mfhd2 has an 004 that is the 001 of bib1.
This mfhd is not held.

Then, we have a subsequent brand new record, mfhd3, that has an 004 that is the 001 of bib1.    no change was made to bib1, so it was not reproduced.  just mfhd3 is processed, but it got held, no knowing/finding that bib1 is already there.   

The new record seems to get held, but we didn't reproduce this, this was from several experiences was saw. 

Ralph, is that correct?

**Comment by patrickzurek** [comment]
JIRA Comment by user: Chris Delis (cedelis)
JIRA Timestamp: 2012-03-13 07:37 PM

Comment body:

Randy saidÃ‘

"I believe the following is the scenario:

bib1 processed, along with mfhd2.  mfhd2 has an 004 that is the 001 of bib1.
This mfhd is not held.

Then, we have a subsequent brand new record, mfhd3, that has an 004 that is the 001 of bib1.    no change was made to bib1, so it was not reproduced.  just mfhd3 is processed, but it got held, no knowing/finding that bib1 is already there.  

The new record seems to get held, but we didn't reproduce this, this was from several experiences was saw. 

Ralph, is that correct?"

Randy, 

My Scenario 2 is very similar to this and it worked correctly in MST.

Here are my Scenarios 1 & 2, again:

Scenario 1: I harvested 4 total records: 2 bib, 2 mfhd.  Each mfhd had a different parent bib (mfhd1's parent was bib1; mfhd2's parent was bib2).  I modified mfhd2's parent so that it was now bib1.  The MST handled this relationship change correctly.  (No held records)

Scenario 2: Follow initial harvest as Scenario 1.  Instead of modifying mfhd2, instead I added a new mfhd3 with a parent of bib1.  The MST handled this new record (and its relationship) correctly.  (No held records) 


Is there a way to obtain records from Sibley?  Not sure how to proceed unless/until we are able to reproduce this in-house.

Chris

**Comment by patrickzurek** [comment]
JIRA Coment by user: admin
JIRA Timestamp: 2012-03-14 02:11 PM

Comment body:

Chris,

I'm checking to see if we can export just Sibley records. Do you need just a few of all of them?

Ralph,

If you can pinpoint only the relevant records, then that would be optimal. Otherwise, I can work with the whole set, if that's an option. 

Thanks,
Chris

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-03-20 02:38 PM

Comment body:

I asked Marcy if she could find any of these, or recreate.  Below is her response.  Chris, sorry to have have wasted time on this.  In the future, we will recreate the bug using records internally before opening an issue for it, and I should have done that this time.    I am closing this.



From: Strong, Marcy 
Sent: Monday, March 19, 2012 2:19 PM
To: Cook, Randall
Subject: Sibley MFHDs

Hi Randy,

The held file is much smaller than it used to be! I couldnâ€™t find any of the examples I found last month, so perhaps itâ€™s no longer an issue? I did go through my email and found some examples I sent Jennifer, when she asked:

Bib: 4779360 (held MFHD: 5382053)
Bib: 4774668 (held MFHD: 5382055)
Bib: 4774667 (held MFHD: 5382903)
Bib: 4772609 (held MFHD: 5383093) 

However, I couldnâ€™t find any current examples in the Held file.

Marcy
